debezium-scripting/debezium-scripting/src/main/java/io/debezium/transforms/scripting/wasm/ChicoryEngine.java

```diff
@@ -189,17 +196,97 @@ public int getString(int proxyObjectRef) {
     }
 
     @WasmExport
-    public int getInt(int proxyObjectRef) {
+    public int getBool(int proxyObjectRef) {
+        var proxyObject = objects.get(proxyObjectRef);
+        if (proxyObject.getClass().isAssignableFrom(Boolean.class)) {
```

Comment:
I wonder if we can use pattern matching here and at the other places instead of reflection.WDYT?

Reply:
Changed to use `instanceof`, in this case the pattern matching is going to be always on a single element, so, an `if` is probably still the preferred construct for readability.
Not fussy on this detail though, if you prefer a different encoding just let me know :+1: